packages/theia-integration/src/node/glsp-socket-server-contribution.ts

```diff
@@ -50,7 +55,8 @@ export namespace GLSPSocketServerContributionOptions {
             ...GLSPServerContributionOptions.createDefaultOptions(),
             socketConnectionOptions: {
                 port: NaN
-            }
+            },
+            webSocketPath: getWebSocketPath()
```

Comment:
If we introduce dedicated launchers for socket /websocket we could do this flag check in the backend-module and bind the right contribution based on the given parameter set. Similiar to how you do it for  the direct websocket usecase.